src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,145 @@
+import React, { useEffect, useMemo } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, PositionRotation } from "../../constants/interfaces";
+import { formatFloatForDisplay } from "../../util";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+interface MetadataPanelProps {
+    selectedAgent: AgentMetadata;
+    uiDisplayData: AgentDisplayNode[];
+}
+
+const MetadataPanel: React.FC<MetadataPanelProps> = ({
+    selectedAgent,
+    uiDisplayData,
+}) => {
+    const [panelExpanded, setPanelExpanded] = React.useState(false);
+
+    const agentSelected = !!selectedAgent.uniqueId;
+    const selectedAgentNotRendered =
+        selectedAgent.agentType >= uiDisplayData.length;
+
+    useEffect(() => {
+        if (!agentSelected) {
+            setPanelExpanded(false);
+        } else {
+            setPanelExpanded(true);
+        }
+    }, [selectedAgent.uniqueId]);
+
+    const MetadataLabels: Record<keyof AgentMetadata, JSX.Element> = {
+        uniqueId: <span>Unique ID</span>,
+        agentType: <span>Agent Type</span>,
+        position: <span>Position</span>,
+        rotation: (
+            <>
+                Rotation
+                <br />
+                (degrees)
+            </>
+        ),
+        radius: <span>Radius</span>,
+    };
+
+    const getFormattedValue = (
+        key: string,
+        value: number | PositionRotation
+    ): JSX.Element | string => {
+        if (selectedAgentNotRendered) {
+            return "-";
+        }
+        if (key === "agentType") {
+            return uiDisplayData[selectedAgent.agentType].title;
+        }
+        if (
+            typeof value === "object" &&
+            (key === "position" || key === "rotation")
+        ) {
+            return (
+                <>
+                    <div> x = {formatFloatForDisplay(value.x)}</div>
+                    <div> y = {formatFloatForDisplay(value.y)}</div>
+                    <div> z = {formatFloatForDisplay(value.z)}</div>
```

Comment:
any intention to put units with these numbers some day?

Reply:
Seems useful, wasn't in design